File: dev/test_data/main/java/expected/implementation_specific/input/snippets/Types/Box/resolve_color.java
/**
 * @param fallback used if the color has not been set
 * @return the color if set, or the fallback otherwise.
 */
public Color resolveColor(Optional<Color> fallback) {
  if (color != null) {
    return color;
  }

  return fallback.orElse(Color.RED);
}

File: dev/test_data/main/typescript/expected/implementation_specific/input/snippets/Types/Box/resolve_color.ts
/**
 * @param fallback - used if {@link color} has not been set
 * @returns {@link color} if set, or the `fallback` otherwise.
 */
resolveColor(fallback: Color | null): Color {
    if (this.color !== null) {
        return this.color;
    }

    return (fallback !== null) ? fallback : Color.Red;
}
